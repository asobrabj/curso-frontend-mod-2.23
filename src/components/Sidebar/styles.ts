import styled from 'styled-components'
import { ContainerTheme } from '../ChangeTheme/styles'
import { NewAnotation } from '../NewAnotation/styles'
import { Input } from '../Search/styles'

export const Container = styled.nav`
  border-right: 1px solid ${(props) => props.theme.color.shadow};
  width: 240px;
  height: 100%;
  text-align: center;
`

export const ContainerExtras = styled.div`
  display: none;
  position: relative;
  height: 150px;

  > ${Input} {
    margin-top: 10px;
    margin-left: 0;
    width: 200px;
  }

  > ${NewAnotation} {
    margin: 10px 0;
    top: 50px;
    left: 20px;
    padding: 2px;
  }

  > ${ContainerTheme} {
    margin-left: 20px;
    margin-top: 60px;
  }

  @media (max-width: 820px) {
    display: block;
  }

  border-bottom: 1px solid ${(props) => props.theme.color.shadow};
`
