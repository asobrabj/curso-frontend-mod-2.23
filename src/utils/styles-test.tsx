import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from '../styles/darkTheme'

export const renderTheme = (Children: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}> {Children} </ThemeProvider>)
}
