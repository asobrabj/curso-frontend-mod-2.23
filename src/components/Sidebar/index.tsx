import dataList from '../../model/list'
import ChangeTheme from '../ChangeTheme'
import List from '../List'
import NewAnotation from '../NewAnotation'
import Search from '../Search'
import * as S from './styles'

interface ITheme {
  isDark: boolean
  changeTheme: () => void
}

export default function Sidebar({ changeTheme, isDark }: ITheme) {
  return (
    <S.Container data-testid="sidebar">
      {dataList.map((item, index) => (
        <List key={index} {...item} />
      ))}
      <S.ContainerExtras>
        <Search />
        <NewAnotation />
        <ChangeTheme changeTheme={changeTheme} isDark={isDark} />
      </S.ContainerExtras>
    </S.Container>
  )
}
