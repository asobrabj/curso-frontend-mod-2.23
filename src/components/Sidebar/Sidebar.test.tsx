import { screen } from '@testing-library/react'
import Sidebar from '.'
import dataList from '../../model/list'
import { renderTheme } from '../../utils/styles-test'

jest.mock('../../context/localStorage/localStorage', () => ({
  ...jest.requireActual('../../context/localStorage/localStorage'),
  useStorage: () => ({
    getFilters: jest.fn(),
  }),
}))

describe('Sidebar Component', () => {
  const mockChangeTheme = jest.fn()

  it('deve renderizar corretamente', () => {
    renderTheme(<Sidebar changeTheme={mockChangeTheme} isDark={false} />)

    const sidebarElement = screen.getByTestId('sidebar')
    expect(sidebarElement).toBeInTheDocument()

    dataList.forEach((item) => {
      const listElement = screen.getByText(item.name)
      expect(listElement).toBeInTheDocument()
    })

    const searchElement = screen.getByPlaceholderText('filtrar anotações')
    expect(searchElement).toBeInTheDocument()

    const newAnotationElement = screen.getByTitle('new-anotation')
    expect(newAnotationElement).toBeInTheDocument()

    const changeThemeElement = screen.getByTitle(/icon theme/i)
    expect(changeThemeElement).toBeInTheDocument()
  })

  it('snapshot', () => {
    const { container } = renderTheme(
      <Sidebar changeTheme={mockChangeTheme} isDark={false} />
    )

    expect(container).toMatchSnapshot()
  })
})
