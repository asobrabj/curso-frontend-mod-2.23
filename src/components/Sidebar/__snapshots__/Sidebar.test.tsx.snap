// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component snapshot 1`] = `
.c12 {
  cursor: pointer;
  margin-left: 450px;
  margin-top: 5px;
  margin-right: 80px;
  position: relative;
  border: 1px solid #999;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.c12 >i {
  position: absolute;
  top: 6px;
  left: 6px;
  font-size: 18px;
  border-radius: 50%;
  color: #bdbbbb;
}

.c2 {
  padding: 16px;
  font-size: 14px;
  font-weight: 400;
  border-bottom: 1px solid #999;
  display: block;
  transition: opacity 1s ease-in-out;
}

.c3 {
  padding: 16px;
  font-size: 14px;
  font-weight: 400;
  border-bottom: 1px solid #999;
  display: none;
  transition: opacity 1s ease-in-out;
}

.c1 {
  text-transform: capitalize;
  cursor: pointer;
}

.c1 >.paragraph {
  display: flex;
  justify-content: center;
  font-size: 18px;
  font-weight: bold;
  padding: 16px 0;
  border-bottom: 1px solid #999;
}

.c1 >.paragraph >i {
  margin-right: 5px;
}

.c8 {
  position: absolute;
  top: 10px;
  right: 70px;
  display: flex;
  align-items: center;
  cursor: pointer;
  padding: 8px;
}

.c9 {
  font-weight: bold;
  margin-right: 8px;
}

.c10 {
  font-size: 24px;
  margin-top: 5px;
}

.c6 {
  margin-left: 120px;
  border: 1px solid;
  border-radius: 8px;
  width: 300px;
  padding: 8px;
  font-size: 16px;
  font-weight: bold;
}

.c6::placeholder {
  color: #999;
}

.c0 {
  border-right: 1px solid #999;
  width: 240px;
  height: 100%;
  text-align: center;
}

.c4 {
  display: none;
  position: relative;
  height: 150px;
  border-bottom: 1px solid #999;
}

.c4 >.c5 {
  margin-top: 10px;
  margin-left: 0;
  width: 200px;
}

.c4 >.c7 {
  margin: 10px 0;
  top: 50px;
  left: 20px;
  padding: 2px;
}

.c4 >.c11 {
  margin-left: 20px;
  margin-top: 60px;
}

@media (max-width: 820px) {
  .c12 {
    margin-left: 226px;
  }
}

@media (max-width: 820px) {
  .c4 {
    display: block;
  }
}

<div>
   
  <nav
    class="c0"
    data-testid="sidebar"
  >
    <div
      class=""
    >
      <ul
        class="c1"
      >
        <li
          class="c2 paragraph"
        >
          <i>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
              />
            </svg>
          </i>
          categoria
        </li>
        <li
          class="c3"
        >
          finanças
        </li>
        <li
          class="c3"
        >
          comida
        </li>
        <li
          class="c3"
        >
          estudo
        </li>
        <li
          class="c3"
        >
          trabalho
        </li>
      </ul>
    </div>
    <div
      class=""
    >
      <ul
        class="c1"
      >
        <li
          class="c2 paragraph"
        >
          <i>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
              />
            </svg>
          </i>
          prioridade
        </li>
        <li
          class="c3"
        >
          urgente
        </li>
        <li
          class="c3"
        >
          alta
        </li>
        <li
          class="c3"
        >
          media
        </li>
        <li
          class="c3"
        >
          baixa
        </li>
      </ul>
    </div>
    <div
      class="c4"
    >
      <input
        class="c5 c6"
        name="search"
        placeholder="filtrar anotações"
        type="text"
        value=""
      />
      <div
        class="c7 c8"
        title="new-anotation"
      >
        <p
          class="c9"
        >
          Nova Anotação
        </p>
        <i
          class="c10"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Circle_Plus"
            >
              <g>
                <path
                  d="M15,12.5H12.5V15a.5.5,0,0,1-1,0V12.5H9a.5.5,0,0,1,0-1h2.5V9a.5.5,0,0,1,1,0v2.5H15A.5.5,0,0,1,15,12.5Z"
                />
                <path
                  d="M12,21.932A9.934,9.934,0,1,1,21.932,12,9.944,9.944,0,0,1,12,21.932ZM12,3.065A8.934,8.934,0,1,0,20.932,12,8.944,8.944,0,0,0,12,3.065Z"
                />
              </g>
            </g>
          </svg>
        </i>
      </div>
      <div
        class="c11 c12"
        data-testid="change theme"
        title="change theme"
      >
        <i
          title="icon theme"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M264 480A232 232 0 0132 248c0-94 54-178.28 137.61-214.67a16 16 0 0121.06 21.06C181.07 76.43 176 104.66 176 136c0 110.28 89.72 200 200 200 31.34 0 59.57-5.07 81.61-14.67a16 16 0 0121.06 21.06C442.28 426 358 480 264 480z"
            />
          </svg>
        </i>
      </div>
    </div>
  </nav>
   
</div>
`;
